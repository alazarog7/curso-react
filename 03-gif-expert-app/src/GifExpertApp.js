import React, { useState } from 'react'
import AddCategory from './Components/AddCategory';
import GifGrid from './Components/GifGrid';

const GifExpertApp = ({defaultCategories = [] })=>{


    const [category, setcategory] = useState(defaultCategories);

    const addCategory = (v)=>{
        setcategory([v, ...category ]);
    };

    return (
        <>
            
            <h2>GifExpertApp</h2>
            <hr/>  
            <AddCategory setcategory = {addCategory} />
            <ol>
                    {
                        category.map((c)=><GifGrid key={c} category={c} />)
                    }
            </ol>
        </>
    );
};

export default GifExpertApp;