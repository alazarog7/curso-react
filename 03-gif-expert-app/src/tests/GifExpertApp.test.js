import React from 'react';
import shallow from "enzyme/build/shallow"
import GifExpertApp from "../GifExpertApp";


describe('<GiffExpertApp/>', () => {

    const categories = ['Homero', 'OK'];
    test('should return', () => {
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
    })

    test('should return a list of categories', () => {
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        
        expect(wrapper.find('GifGrid').length).toBe(categories.length)


    })
    
    
})
