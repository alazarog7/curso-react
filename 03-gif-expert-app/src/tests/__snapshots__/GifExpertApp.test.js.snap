// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiffExpertApp/> should return 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setcategory={[Function]}
  />
  <ol>
    <GifGrid
      category="Homero"
      key="Homero"
    />
    <GifGrid
      category="OK"
      key="OK"
    />
  </ol>
</Fragment>
`;
